### ISSUE TITLE
KopfRunner import fails in Python 3.11

### ISSUE THREAD

**Issue by jacobtomlinson** [nolar/kopf#972]
### Long story short

I'm just starting to play with Python 3.11 support in a project I maintain that uses `kopf`. I added [3.11 to our CI in a PR](https://github.com/dask/dask-kubernetes/pull/600) but we don't seem to get past `from kopf.testing import KopfRunner`.

### Kopf version

1.35.6

### Kubernetes version

all

### Python version

3.11

### Code

```python
from kopf.testing import KopfRunner
```

### Logs

```python-traceback
 ImportError while loading conftest '/home/runner/work/dask-kubernetes/dask-kubernetes/dask_kubernetes/conftest.py'.
dask_kubernetes/conftest.py:8: in <module>
    from kopf.testing import KopfRunner
/opt/hostedtoolcache/Python/3.11.0/x64/lib/python3.11/site-packages/kopf/__init__.py:10: in <module>
    from kopf import (
/opt/hostedtoolcache/Python/3.11.0/x64/lib/python3.11/site-packages/kopf/on.py:18: in <module>
    from kopf._core.intents import callbacks, causes, filters, handlers, registries
/opt/hostedtoolcache/Python/3.11.0/x64/lib/python3.11/site-packages/kopf/_core/intents/causes.py:204: in <module>
    @dataclasses.dataclass
/opt/hostedtoolcache/Python/3.11.0/x64/lib/python3.11/dataclasses.py:1221: in dataclass
    return wrap(cls)
/opt/hostedtoolcache/Python/3.11.0/x64/lib/python3.11/dataclasses.py:1211: in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash,
/opt/hostedtoolcache/Python/3.11.0/x64/lib/python3.11/dataclasses.py:959: in _process_class
    cls_fields.append(_get_field(cls, name, type, kw_only))
/opt/hostedtoolcache/Python/3.11.0/x64/lib/python3.11/dataclasses.py:816: in _get_field
    raise ValueError(f'mutable default {type(f.default)} for field '
E   ValueError: mutable default <class 'kopf._cogs.structs.diffs.Diff'> for field diff is not allowed: use default_factory
```

**Comment by nolar** [comment]
Thanks for reporting. Yes, this was fixed in a recent PR #968. Perhaps, let me make a new release candidate — it seems like a good time for this.

Done. [Kopf 1.36.0](https://github.com/nolar/kopf/releases/tag/1.36.0) should fix the issue and enable Python 3.11 in general. Can you please try it?

**Comment by jacobtomlinson** [comment]
Sorry should've searched in PRs as well as issues before opening this.

That gets my CI past the `kopf` error, many thanks! Now onto the next one 😄.
